features/netsocket/NetworkInterface.h

```diff
@@ -324,6 +324,18 @@ class NetworkInterface: public DNS {
      */
     virtual nsapi_error_t add_dns_server(const SocketAddress &address, const char *interface_name);
 
+    /** Get a domain name server from a list of servers to query
+     *
+     *  Returns a DNS server address for a index. If returns error no more
+     *  DNS servers to read.
+     *
+     *  @param index    Index of the DNS server, starts from zero
+     *  @param address  Destination for the host address
+     *  @param interface_name  Network interface name
+     *  @return         NSAPI_ERROR_OK on success, negative error code on failure
+     */
+    virtual nsapi_error_t get_dns_server(int index, SocketAddress *address, const char *interface_name = NULL);
```

Comment:
Looking at the file, it was already changed to contain "override" rather than "virtual", please  review the other methods and update this one as well

cc @kjbracey-arm was the design Mbed OS guidance updated and team as well to follow the latest changes?

Reply:
This is a base class declaration so this needs to be virtual.

Reply:
Correct, I had to opened a different file , can;t find it now.